ChatSX/components/ParseDeepLink/index.ts

```diff
@@ -0,0 +1,101 @@
+import {myClient} from '../../..';
+import {PATH_REGEX, QUERY_REGEX, VALID_URI_REGEX} from '../../constants/Regex';
+import {initAPI} from '../../store/actions/homefeed';
+
+interface DeepLinkRequest {
+  uri: string;
+  userName: string;
+  uuid: string;
+  isGuest: boolean;
+}
+
+interface DeepLinkResponse {
+  success: boolean;
+  errorMessage?: string;
+}
+
+// this function is to check URI is valid or not
+function isValidURI(uri: string): boolean {
+  const regex = VALID_URI_REGEX;
+  return regex.test(uri);
+}
+
+// this function is to parse deep link url
+async function parseDeepLink(
+  request: DeepLinkRequest,
+  responseCallback?: (response: DeepLinkResponse) => void,
+) {
+  const uri = request.uri;
+
+  if (isValidURI(uri)) {
+    const pathRegex = PATH_REGEX;
```

Comment:
this is unnecessary variable `PATH_REGEX` can directly be used for further operations